src/fides/api/ops/service/connectors/saas/connector_registry_service.py

```diff
@@ -20,65 +18,75 @@
+    load_yaml_as_string,
+    replace_config_placeholders,
+    replace_dataset_placeholders,
 )
 
-_registry: Optional[ConnectorRegistry] = None
-registry_file = "data/saas/saas_connector_registry.toml"
 
+class ConnectorTemplateLoader(ABC):
+    @abstractmethod
+    def get_connector_templates(self) -> Dict[str, ConnectorTemplate]:
+        """Returns a map of connection templates"""
 
-class ConnectorTemplate(BaseModel):
+
+class FileConnectorTemplateLoader(ConnectorTemplateLoader):
     """
-    A collection of paths to artifacts that make up
-    a complete SaaS connector (SaaS config, dataset, etc.)
+    Loads SaaS connector templates from the data/saas directory.
     """
 
-    config: str
-    dataset: str
-    icon: str
-    human_readable: str
-
-    @validator("config")
-    def validate_config(cls, config: str) -> str:
-        """Validates the config at the given path"""
-        SaaSConfig(**load_config(config))
-        return config
+    def __init__(self) -> None:
+        self.templates: Dict[str, ConnectorTemplate] = {}
+        for file in os.listdir("data/saas/config"):
+            if file.endswith(".yml"):
+                config_file = os.path.join("data/saas/config", file)
+                config_dict = load_config(config_file)
+                connector_type = config_dict["type"]
+                human_readable = config_dict["name"]
 
-    @validator("dataset")
-    def validate_dataset(cls, dataset: str) -> str:
-        """Validates the dataset at the given path"""
-        Dataset(**load_dataset(dataset)[0])
-        return dataset
+                try:
+                    icon = encode_file_contents(f"data/saas/icon/{connector_type}.svg")
+                except FileNotFoundError:
+                    icon = encode_file_contents("data/saas/icon/default.svg")
 
-    @validator("icon")
-    def validate_icon(cls, icon: str) -> str:
-        """Validates the icon at the given path"""
-        if not exists(icon):
-            raise ValueError(f"Icon file {icon} was not found")
-        return icon
+                # store connector template for retrieval
+                try:
+                    self.templates[connector_type] = ConnectorTemplate(
+                        config=load_yaml_as_string(config_file),
+                        dataset=load_yaml_as_string(
+                            f"data/saas/dataset/{connector_type}_dataset.yml"
+                        ),
+                        icon=icon,
+                        human_readable=human_readable,
+                    )
+                except Exception:
+                    logger.exception("Unable to load {} connector", connector_type)
 
+    def get_connector_templates(self) -> Dict[str, ConnectorTemplate]:
+        return self.templates
 
-class ConnectorRegistry(BaseModel):
-    """A map of SaaS connector templates"""
 
-    __root__: Dict[str, ConnectorTemplate]
+class ConnectorRegistry:
+    loader = FileConnectorTemplateLoader()
```

Comment:
That's exactly it, here's what I ran ahead with in my other branch. I hard-coded the loaders instead of iterating through available loaders because I want to control the order, meaning that if a connector is defined in the database it should override a connector read from the `data/saas` directory.

```
class ConnectorRegistry:

    _instance = None
    _templates: Dict[str, ConnectorTemplate] = {}

    @classmethod
    def get_instance(cls) -> "ConnectorRegistry":
        if cls._instance is None:
            cls._instance = cls()
            cls._instance._templates = {
                **FileConnectorTemplateLoader().get_connector_templates(),
                **CustomConnectorTemplateLoader().get_connector_templates(),
            }
        return cls._instance

    @classmethod
    def connector_types(cls) -> List[str]:
        """List of registered SaaS connector types"""
        return list(cls.get_instance()._templates.keys())

    @classmethod
    def get_connector_template(cls, connector_type: str) -> Optional[ConnectorTemplate]:
        """
        Returns an object containing the various SaaS connector artifacts
        """
        return cls.get_instance()._templates.get(connector_type)
```

Reply:
hmmm. i've got a few questions on the `get_instance()` choice, but we can dig into that on the subsequent PR, no need to hold things up here - what you've got so far on this PR specifically looks good to me!